### PR TITLE
feat: add an option to make a link from Tab - FE-3486

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
image

### Current behaviour
There is no possibility to make a link from Tab

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### Testing instructions

Storybook chromatic:
https://5ecf782fe724630022d27d7d-dqrmncmmhi.chromatic.com/?path=/story/design-system-tabs--with-link-as-a-tab

Storybook local:
story/design-system-tabs--with-link-as-a-tab

### CODE REVIEW

**Comment by bulaj** [comment]
image
Isn't it a little misaligned vertically? macOs safair, on chrome looks the same

File: src/components/tabs/__internal__/tab-title/tab-title.component.js [hunk 1]
```diff
@@ -104,8 +125,12 @@ const TabTitle = React.forwardRef(
         isInSidebar={isInSidebar}
         {...tabTitleProps}
         {...tagComponent("tab-header", tabTitleProps)}
+        onKeyDown={handleKeyDown}
       >
         <StyledTitleContent
+          as={href && "a"}
```
Comment (bulaj) [inline src/components/tabs/__internal__/tab-title/tab-title.component.js:131]: Initialize ``const hasHref = !!href`` at the top to make it a litttle more verbose and avoid possible xss.

**Review by bulaj — CHANGES_REQUESTED** [review]

File: src/components/tabs/__internal__/tab/tab.d.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export interface TabProps {
   infoMessage: string;
   siblings?: React.ReactNode[];
   titlePosition?: 'before' | 'after';
+  href: string;
```
Comment (edleeks87) [inline src/components/tabs/__internal__/tab/tab.d.ts:19]:
```suggestion
  href?: string;
```

**Review by edleeks87 — COMMENTED** [review]
Not much else to add past what @bulaj has raised already, happy to approve once they've been addressed

**Comment by edleeks87** [comment]
While you're updating this component @dawidzarzycki, could you move the `tab` directory, `tabs.component`, `tabs.d.ts`, `tabs.stories.js`, `tabs.spec.js`, `tabs.stories.mdx` and `tabs.style.js` out of the `__internal__` folder as well please? We've changed our planned approach since they were moved there originally.

**Review by bulaj — APPROVED** [review]

**Review by bulaj — APPROVED** [review]

File: src/components/tabs/tabs.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import Tab from './tab';
+import Tab from '../tab';
```
Comment (edleeks87) [inline src/components/tabs/tabs.d.ts:2]:
```suggestion
import Tab from './tab';
```

**Review by edleeks87 — DISMISSED** [review]

**Review by bulaj — DISMISSED** [review]

**Review by bulaj — DISMISSED** [review]

File: cypress/features/regression/test/tabs.feature [hunk 4]
```diff
@@ -12,4 +12,9 @@ Feature: Tabs component
       | 2  |
       | 3  |
       | 4  |
-      | 5  |
\ No newline at end of file
+      | 5  |
+
+  @positive
+  Scenario: Check that Tab has link properties
+    Given I open "Design System Tabs" component page "with-link as a tab" in no iframe
```
Comment (DlgSHi) [inline cypress/features/regression/test/tabs.feature:19]:
```suggestion
    Given I open "Design System Tabs" component page "with link as a tab" in no iframe
```

**Review by DlgSHi — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by bulaj — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**